corehq/apps/app_manager/static/app_manager/js/managed_app.js

```diff
@@ -1,68 +1,58 @@
-$(function () {
-    var initial_page_data = hqImport('hqwebapp/js/initial_page_data').get,
-        init = hqImport('app_manager/js/app_manager').init,
-        app = initial_page_data('app_subset');
+hqDefine("app_manager/js/managed_app", function() {
```

Comment:
It's a bit unintuitive for importing a module to affect DOM state—that's kind of exactly the coupling we're trying to avoid by modularizing things.

I'd suggest replacing this `$(function () {...})` with an

``` return {
    initAppManager: function () {
         ...
    }
```

and then in the html have a short piece of inline js:

```
$(function () {
    hqImport("app_manager/js/managed_app").initAppManager();
});
```

Additionally, if it feels right, it could be nice to pass in the `$("#element_id")` expressions rather than hardcoding them here, though that I'm less particular about.

Reply:
I'm not quite following, how are you imagining this integrating with RequireJS? I've been aiming to get that kind of initialization js *out* of HTML.

Our modules mostly fall into one of two categories: reusable (utilities, widgets, etc.) and page-specific. Page-specific modules like this one also serve as the entry points in RequireJS environments (each a page names a "main" module with the `requirejs_main` tag, and all dependencies for that page flow from that module), and they aren't really meant to be imported elsewhere.

> It's a bit unintuitive for importing a module to affect DOM state

That's not obvious to me - if a module's job is to control a particular page, seems plausible that it also initialize the page. I'd be open to talking about distinguishing between these types of modules, but the horse is pretty far out of the barn when it comes to having document ready handlers inside of modules.

Reply:
(it's possible we've had this conversation before, but I'm not sure. happy to talk through it either way.)

Reply:
Okay, that response makes a lot of sense and is also super familiar. Pretty sure I bugged you about this before and was convinced by the same response.

I do think eventually it will be nice to distinguish entry-point modules (which execute functionality by virtue of just being included in a script tag) from normal modules (which don't _do_ anything when included in a script tag except make their methods/properties available through `hqImport`). Ideally, even some page-specific modules would expose functionality rather then execute it, and then the entry-point module would import from those and execute.

I have some more thoughts, but maybe I'll offline with you. I'm not too tied to a position, and I think you have a lot more context than I do, but I think it'd be worth talking through.